cmd/crank/update.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/alecthomas/kong"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	v1 "github.com/crossplane/crossplane/apis/pkg/v1"
+	typedclient "github.com/crossplane/crossplane/internal/client/clientset/versioned/typed/pkg/v1"
+	"github.com/crossplane/crossplane/internal/xpkg"
+
+	// Load all the auth plugins for the cloud providers.
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+)
+
+// updateCmd updates a package.
+type updateCmd struct {
+	Configuration updateConfigCmd   `cmd:"" help:"Update a Configuration package."`
+	Provider      updateProviderCmd `cmd:"" help:"Update a Provider package."`
+}
+
+// Run runs the update cmd.
+func (c *updateCmd) Run(b *buildChild) error {
+	return nil
+}
+
+// updateConfigCmd updates a Configuration.
+type updateConfigCmd struct {
+	Package string `arg:"" help:"Image containing Configuration package."`
+
+	Name                 string   `arg:"" optional:"" help:"Name of Configuration."`
+	RevisionHistoryLimit int64    `short:"rl" help:"Revision history limit."`
+	ManualActivation     bool     `short:"m" help:"Enable manual revision activation policy."`
+	PackagePullSecrets   []string `help:"List of secrets used to pull package."`
+}
+
+// Run runs the Configuration update cmd.
+func (c *updateConfigCmd) Run(k *kong.Context) error {
+	rap := v1.AutomaticActivation
+	if c.ManualActivation {
+		rap = v1.ManualActivation
+	}
+	pkgName := c.Name
+	if pkgName == "" {
+		ref, err := name.ParseReference(c.Package)
+		if err != nil {
+			return errors.Wrap(err, errPkgIdentifier)
+		}
+		pkgName = xpkg.ToDNSLabel(ref.Context().RepositoryStr())
+	}
+	packagePullSecrets := make([]corev1.LocalObjectReference, len(c.PackagePullSecrets))
+	for i, s := range c.PackagePullSecrets {
+		packagePullSecrets[i] = corev1.LocalObjectReference{
+			Name: s,
+		}
+	}
+	cr := &v1.Configuration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: pkgName,
+		},
+		Spec: v1.ConfigurationSpec{
+			PackageSpec: v1.PackageSpec{
+				Package:                  c.Package,
```

Comment:
I think the `update` command should only modify the tag. So the command would look something like:

```
kubectl crossplane provider update <name-of-provider-k8s-object> <tag-to-update-spec.package-to>
```

An example would be:

```
kubectl crossplane provider update provider-aws v0.16.0
```

It will take some consideration to handle digest (`@`) vs tag (`:`) correctly, but `github.com/google/go-containerregistry/pkg/name` has some utilities for parsing.

Reply:
I think we can just use `@` if "tag" starts with `sha256` and otherwise use `:`